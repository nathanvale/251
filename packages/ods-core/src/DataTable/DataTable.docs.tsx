import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { propDescriptions } from "../Table/Table.docs";
import { DataTable, DataTableProps } from "./DataTable";

export const docs: ComponentDocs<DataTableProps> = {
  category: "Content",
  componentName: "DataTable",
  description:
    "DataTable is a component that allows for a controlled version of Table. Open the examples in Playroom to see the behaviour of this component.",
  propDescriptions: {
    ...propDescriptions,
    headings: "List of headings for the table.",
    data: "List of lists of content for the table.",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <DataTable
          headings={["1", "2", "3"]}
          data={[
            ["One", "Two", "Three"],
            ["Uno", "Dos", "Tres"],
            ["Een", "Twee", "Drie"],
          ]}
        />
      ),
    },
    additional: [
      {
        label: "Without headings",
        Code: () => (
          <DataTable
            data={[
              ["One", "Two", "Three"],
              ["Uno", "Dos", "Tres"],
              ["Een", "Twee", "Drie"],
            ]}
          />
        ),
      },
    ],
  },
  snippets: [],
};
