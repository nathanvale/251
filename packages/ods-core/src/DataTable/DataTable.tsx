import React, { ReactNode } from "react";
import { Table, TableProps, defaultProps } from "../Table";
import { TableBody } from "../TableBody";
import { TableCell } from "../TableCell";
import { TableHead } from "../TableHead";
import { TableRow } from "../TableRow";

export interface DataTableProps
  extends Pick<TableProps, Exclude<keyof TableProps, "children">> {
  headings?: ReactNode[];
  data?: ReactNode[][];
}

export const DataTable = ({
  headings = [],
  data = [],
  muiProps,
  ...props
}: DataTableProps) =>
  !!headings.length || !!data.length ? (
    <Table {...props} {...(muiProps || {})}>
      {!!headings.length && (
        <TableHead>
          {headings.map((heading, headingIdx) => (
            <TableCell key={headingIdx}>{heading}</TableCell>
          ))}
        </TableHead>
      )}

      <TableBody>
        {data.map((row, rowIdx) => (
          <TableRow key={rowIdx}>
            {row.map((column, colIdx) => (
              <TableCell key={`${rowIdx}-${colIdx}`}>{column}</TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  ) : null;

DataTable.defaultProps = {
  ...defaultProps,
  "data-id": "data-table",
};

DataTable.displayName = "DataTable";
