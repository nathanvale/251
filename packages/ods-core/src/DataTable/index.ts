export { DataTable } from "./DataTable";
export type DataTableProps = import("./DataTable").DataTableProps;
