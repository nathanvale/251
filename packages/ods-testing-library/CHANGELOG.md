# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
