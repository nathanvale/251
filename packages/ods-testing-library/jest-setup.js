import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
