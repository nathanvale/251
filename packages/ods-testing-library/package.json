{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
