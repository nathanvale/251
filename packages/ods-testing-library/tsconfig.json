{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist/typings",
    "rootDir": "./src"
  },
  "include": [
    "src",
    "src/**/*.json",
    "../../globals.d.ts",
    "../../@types",
    "jest-setup.ts"
  ],
  "references": [{"path": "../ods-themes/tsconfig.json"}]
}
