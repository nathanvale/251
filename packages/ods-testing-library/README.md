# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
