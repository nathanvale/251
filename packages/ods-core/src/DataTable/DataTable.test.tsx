import React from "react";
import { render } from "@origin-digital/ods-testing-library";

import { DataTable, DataTableProps } from "./DataTable";

const renderDataTable = (
  dataTableProps: Partial<DataTableProps> = {},
  hideHeadings = false,
  hideData = false
) =>
  render(
    <DataTable
      headings={!hideHeadings ? ["1", "2", "3"] : undefined}
      data={
        !hideData
          ? [
              ["One", "Two", "Three"],
              ["Uno", "Dos", "Tres"],
            ]
          : undefined
      }
      {...dataTableProps}
    />
  );

test("It renders", () => {
  const { container } = renderDataTable();
  expect(container.firstChild).toMatchSnapshot();
});

test("It renders 3 headings", () => {
  const { container } = renderDataTable();

  const tableHead = container.querySelectorAll("thead");
  const tableHeadCells = container.querySelectorAll("th");

  expect(tableHead).toHaveLength(1);
  expect(tableHeadCells).toHaveLength(3);
});

test("It renders 0 headings", () => {
  const { container } = renderDataTable({}, true);

  const tableHead = container.querySelectorAll("thead");
  const tableHeadCells = container.querySelectorAll("th");

  expect(tableHead).toHaveLength(0);
  expect(tableHeadCells).toHaveLength(0);
});

test("It renders 6 cells", () => {
  const { container } = renderDataTable();

  const tableCells = container.querySelectorAll("td");

  expect(tableCells).toHaveLength(6);
});

test("It renders nothing", () => {
  const { container } = renderDataTable({}, true, true);

  const table = container.querySelectorAll("table");

  expect(table).toHaveLength(0);
});
