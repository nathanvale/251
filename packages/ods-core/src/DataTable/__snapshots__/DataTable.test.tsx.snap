// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders 1`] = `
<div
  class="MuiTableContainer-root"
>
  <table
    class="MuiTable-root"
    data-id="data-table"
  >
    <thead
      class="MuiTableHead-root TableHead-whiteBackground-5"
      data-id="table-head"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-7 TableCell-withPadding-12 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-11 TableCell-withFontSize-9 TableCell-withFontSize-13 TableCell-withoutFirstLastPadding-8"
        data-id="table-cell"
        scope="col"
      >
        1
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-7 TableCell-withPadding-15 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-14 TableCell-withFontSize-9 TableCell-withFontSize-16 TableCell-withoutFirstLastPadding-8"
        data-id="table-cell"
        scope="col"
      >
        2
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-7 TableCell-withPadding-18 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-17 TableCell-withFontSize-9 TableCell-withFontSize-19 TableCell-withoutFirstLastPadding-8"
        data-id="table-cell"
        scope="col"
      >
        3
      </th>
    </thead>
    <tbody
      class="MuiTableBody-root"
      data-id="table-body"
    >
      <tr
        class="MuiTableRow-root"
        data-id="table-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-23 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-22 TableCell-withFontSize-9 TableCell-withFontSize-24 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          One
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-26 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-25 TableCell-withFontSize-9 TableCell-withFontSize-27 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          Two
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-29 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-28 TableCell-withFontSize-9 TableCell-withFontSize-30 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          Three
        </td>
      </tr>
      <tr
        class="MuiTableRow-root"
        data-id="table-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-32 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-31 TableCell-withFontSize-9 TableCell-withFontSize-33 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          Uno
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-35 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-34 TableCell-withFontSize-9 TableCell-withFontSize-36 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          Dos
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-7 TableCell-withPadding-38 TableCell-withVerticalAlign-6 TableCell-withVerticalAlign-37 TableCell-withFontSize-9 TableCell-withFontSize-39 TableCell-withoutFirstLastPadding-8"
          data-id="table-cell"
        >
          Tres
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
